src/solver/tron.jl

```diff
@@ -0,0 +1,212 @@
+using LinearOperators
+# A trust-region solver for unconstrained optimization with bounds
+
+export tron
+
+function active(x, l, u; ϵlu = 1e-4)
+  if isa(ϵlu, Float64)
+    ϵlu = (u-l)*ϵlu
+  end
```

Comment:
Is it worth creating a new vector for this? Could we just use `(u[i] - l[i]) * ϵlu` in the loop?